src/cli/peakdetector/PeakDetectorCLI.cpp

```diff
@@ -1070,6 +1070,52 @@ void PeakDetectorCLI::saveCSV(string setName) {
     }
     csvreports->closeFiles();
 
+	// Check whether there's a missing C12 PARENT label in the written CSV file.
+    bool atLeastOneIsotopeLabelIsEmpty = false;
+    bool fileNeedsCorrection = false;
+    vector<int> groupsWithMissingLabels;
+    CsvParser *csvParser = CsvParser_new(fileName.c_str(), ",", 1);
+    CsvRow *currentRow = NULL;
+    // Iterates over each row, continuously assigning a pointer to an unread CsvRow.
+    while (currentRow = CsvParser_getRow(csvParser)) {
+        // eighth field is the isotope label.
+        string isotopeLabel = string(currentRow->fields_[7]);
+        if (isotopeLabel.empty()) {
+            int groupId = atoi(currentRow->fields_[2]);
+            groupsWithMissingLabels.push_back(groupId);
+            atLeastOneIsotopeLabelIsEmpty = true;
+        } else {
```

Comment:
@saifulbkhan both flags signify the same thing, no? as soon as you find a missing label, fileNeedsCorrection goes true. We don't even need the else block

Reply:
actually, we don't need either of the flags. Just check for the groupsWithMissingLabels vector size. If there are elements then we need to correct the file.

Reply:
We will need two flags, albeit there is a simpler way of doing this. I'll make the changes as discussed.